includes/collection/class-actors.php

```diff
@@ -76,6 +76,16 @@ public static function get_by_id( $user_id ) {
 	 * @return User|Blog|Application|WP_Error The Actor or WP_Error if user not found.
 	 */
 	public static function get_by_username( $username ) {
+		// Check if we're dealing with an old domain request.
+		$old_domain     = \get_option( 'activitypub_old_domain', '' );
+		$request_domain = isset( $_SERVER['HTTP_HOST'] ) ? \sanitize_text_field( \wp_unslash( $_SERVER['HTTP_HOST'] ) ) : '';
```

Comment:
What about if we encapsulate this in a function and test also for the referrer? this way, we could also handle domains that point to the same server?

Reply:
Fixed in [698e2ac](https://github.com/Automattic/wordpress-activitypub/pull/1530/commits/698e2ac7e247d26842f83c3a644243fed0e469bb)